src/measures/finite.jl

```diff
@@ -871,6 +871,503 @@ end
+
+"""
+    multiclass_false_negative_rate
+
+$(docstring(MulticlassFalseNegativeRate()))
+
+    multiclass_false_negative_rate(ŷ, y)
+    multiclass_false_negative_rate(ŷ, y, class_w)
+
+False negative rate for multiclass observations `ŷ` and ground truth `y`.
+$DS_MICRO_MACRO `MulticlassFalseNegativeRate(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_false_negative_rate)`.
+
+"""
+const multiclass_false_negative_rate = MulticlassFalseNegativeRate()
+const multiclass_falsenegative_rate  = MulticlassFalseNegativeRate()
+const multiclass_fnr                 = MulticlassFalseNegativeRate()
+const MFNR                           = MulticlassFalseNegativeRate
+const multiclass_miss_rate           = MFNR()
+
+"""
+    multiclass_false_discovery_rate
+
+$(docstring(MulticlassTruePositiveRate()))
+
+    multiclass_false_discovery_rate(ŷ, y)
+    multiclass_false_discovery_rate(ŷ, y, class_w)
+
+False discovery rate for multiclass observations `ŷ` and ground truth `y`.
+$DS_MICRO_MACRO `MulticlassFalseDiscoveryRate(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_false_discovery_rate)`.
+
+"""
+const multiclass_false_discovery_rate = MulticlassFalseDiscoveryRate()
+const multiclass_falsediscovery_rate  = MulticlassFalseDiscoveryRate()
+const multiclass_fdr                  = MulticlassFalseDiscoveryRate()
+const MFDR                            = MulticlassFalseDiscoveryRate
+
+"""
+    multiclass_positive_predictive_value
+
+$(docstring(MulticlassPrecision()))
+
+    multiclass_positive_predictive_value(ŷ, y)
+    multiclass_positive_predictive_value(ŷ, y, class_w)
+
+Positive predictive value for multiclass observations `ŷ` and ground truth
+`y`. $DS_MICRO_MACRO `MulticlassPrecision(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_positive_predictive_value)`.
+
+"""
+const multiclass_precision                 = MulticlassPrecision()
+const multiclass_ppv                       = MulticlassPrecision()
+const multiclass_positive_predictive_value = MulticlassPrecision()
+const multiclass_positivepredictive_value  = MulticlassPrecision()
+const MPPV                                 = MulticlassPrecision
+
+"""
+    multiclass_negative_predictive_value
+
+$(docstring(MulticlassNPV()))
+
+    multiclass_negative_predictive_value(ŷ, y)
+    multiclass_negative_predictive_value(ŷ, y, class_w)
+
+Negative predictive value for multiclass observations `ŷ` and ground truth
+`y`.$DS_MICRO_MACRO `MulticlassNPV(; average=:micro)`. $CLASS_W
+
+For more information, run `info(multiclass_negative_predictive_value)`.
+
+"""
+const multiclass_npv                       = MulticlassNPV()
+const multiclass_negative_predictive_value = MulticlassNPV()
+const multiclass_negativepredictive_value  = MulticlassNPV()
+const MNPV                                 = MulticlassNPV
+
+## INTERNAL FUCNTIONS ON MULTICLASS CONFUSION MATRIX
+
+_mtp(m::CM{N}) where N = diag(m.mat)
```

Comment:
```suggestion
_mtp(m::CM) = diag(m.mat)
```
It is known in julia that using `CM{N}` instead of `CM` slightly affects performance when `N` isn't used in the function definition. Hence the above suggestion.

Reply:
They yield same `@benchmark` performance, I believe both are identical, as CM = ConfusionMatrix{N} where N.

I tried with  `CM_ = ConfusionMatrix`, which slightly performs worse.

```
mfp(m::CM{N}) where N = sum(m.mat, dims=2) .- diag(m.mat)
_mfp2(m::CM_) = sum(m.mat, dims=2) .- diag(m.mat)

julia> @benchmark MLJBase._mfp2(cm)
BenchmarkTools.Trial: 
  memory estimate:  368 bytes
  allocs estimate:  8
  --------------
  minimum time:     781.610 ns (0.00% GC)
  median time:      799.080 ns (0.00% GC)
  mean time:        861.941 ns (4.63% GC)
  maximum time:     52.653 μs (97.38% GC)
  --------------
  samples:          10000
  evals/sample:     100

julia> @benchmark MLJBase._mfp(cm)
BenchmarkTools.Trial: 
  memory estimate:  368 bytes
  allocs estimate:  8
  --------------
  minimum time:     778.592 ns (0.00% GC)
  median time:      794.233 ns (0.00% GC)
  mean time:        851.195 ns (4.51% GC)
  maximum time:     50.301 μs (97.59% GC)
  --------------
  samples:          10000
  evals/sample:     103
```

Reply:
You can stick to using ` CM = ConfusionMatrix{N} where N`

Reply:
However, I did drop `where N` wherever the code could be made compact, later when noticed.

Reply:
Again the code for bench-marking should be
```julia
julia> @benchmark MLJBase._mfp2($cm)
julia> @benchmark MLJBase._mfp($cm)
```